server/service/mdm.go

```diff
@@ -1349,19 +1350,39 @@ func (svc *Service) validateProfileLabels(ctx context.Context, labelNames []stri
 		return nil, ctxerr.Wrap(ctx, err, "getting label IDs by name")
 	}
 
-	if len(labels) != len(labelNames) {
+	uniqueNames := make(map[string]bool)
+	for _, entry := range labelNames {
+		if _, value := uniqueNames[entry]; !value {
+			uniqueNames[entry] = true
+		}
+	}
+
+	if len(labels) != len(uniqueNames) {
 		return nil, &fleet.BadRequestError{
 			Message:     "some or all the labels provided don't exist",
-			InternalErr: fmt.Errorf("len(labels) != len(labelNames), names provided: %v", labelNames),
+			InternalErr: fmt.Errorf("names provided: %v", labelNames),
 		}
 	}
 
-	var profLabels []fleet.ConfigurationProfileLabel
+	profLabels := make(map[string]fleet.ConfigurationProfileLabel)
 	for labelName, labelID := range labels {
-		profLabels = append(profLabels, fleet.ConfigurationProfileLabel{
+		profLabels[labelName] = fleet.ConfigurationProfileLabel{
 			LabelName: labelName,
 			LabelID:   labelID,
-		})
+		}
+	}
+	return profLabels, nil
+}
+
+func (svc *Service) validateProfileLabels(ctx context.Context, labelNames []string) ([]fleet.ConfigurationProfileLabel, error) {
+	labelMap, err := svc.batchValidateProfileLabels(ctx, labelNames)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "validating single profile")
```

Comment:
Just to make it a bit more specific:

```suggestion
		return nil, ctxerr.Wrap(ctx, err, "validating profile labels")
```